ci/scan_flattened_deps.py

```diff
@@ -0,0 +1,343 @@
+old_deps = []  # deps which have not been updated in more than 1 year
+
+sarif_log = {
+    '$schema':
+        'https://json.schemastore.org/sarif-2.1.0.json', 'version':
+            '2.1.0', 'runs': [{
+                'tool': {'driver': {'name': 'OSV Scan', 'rules': []}},
+                'results': []
+            }]
+}
+
+
+def sarif_result():
+  """
+  Returns the template for a result entry in the Sarif log,
+  which is populated with CVE findings from OSV API
+  """
+  return {
+      'ruleId':
+          'N/A', 'message': {'text': 'OSV Scan Finding'}, 'locations': [{
+              'physicalLocation': {
+                  'artifactLocation': {
+                      'uri': 'No location associated with this finding'
+                  },
+                  'region': {'startLine': 1, 'startColumn': 1, 'endColumn': 1}
+              }
+          }]
+  }
+
+
+def sarif_rule():
+  """
+  Returns the template for a rule entry in the Sarif log,
+  which is populated with CVE findings from OSV API
+  """
+  return {
+      'id': 'OSV Scan', 'name': 'OSV Scan Finding',
+      'shortDescription': {'text': 'Insert OSV id'}, 'fullDescription': {
+          'text': 'Vulnerability found by scanning against the OSV API'
+      }, 'help': {
+          'text':
+              'More details in the OSV DB at: https://osv.dev/vulnerability/'
+      }, 'defaultConfiguration': {'level': 'error'}, 'properties': {
+          'problem.severity': 'error', 'security-severity': '9.8',
+          'tags': ['supply-chain', 'dependency']
+      }
+  }
+
+
+def parse_deps_file(deps_flat_file):
+  """
+  Takes input of fully qualified dependencies,
+  for each dep find the common ancestor commit SHA
+  from the upstream and query OSV API using that SHA
+
+  If the commit cannot be found or the dep cannot be
+  compared to an upstream, prints list of those deps
+  """
+  queries = []  # list of queries to submit in bulk request to OSV API
+  deps = open(deps_flat_file, 'r')
+  lines = deps.readlines()
+
+  headers = {
+      'Content-Type': 'application/json',
+  }
+  osv_url = 'https://api.osv.dev/v1/querybatch'
+
+  os.mkdir(
+      CHECKOUT_ROOT + '/clone-test'
+  )  #clone deps with upstream into temporary dir
+
+  # Extract commit hash, save in dictionary
+  for line in lines:
+    os.chdir(CHECKOUT_ROOT + '/clone-test')
+    dep = line.strip().split(
+        '@'
+    )  # separate fully qualified dep into name + pinned hash
+
+    common_commit = get_common_ancestor_commit(dep)
+    if common_commit is not None:
+      queries.append({'commit': common_commit})
+    else:
+      failed_deps.append(dep[0])
+
+  print(
+      'Dependencies that could not be parsed for ancestor commits: ' +
+      ', '.join(failed_deps)
+  )
+  print(
+      'Dependencies that have not been rolled in at least 1 year: ' +
+      ', '.join(old_deps)
+  )
+
+  # Query OSV API using common ancestor commit for each dep
+  # return any vulnerabilities found
+  responses = requests.post(
```

Comment:
This is the only usage I'm seeing. Can't this be replaced with a call from a library that's already available?